src/main/common/components/payment-intention/model/paymentPlan.ts

```diff
@@ -60,7 +67,20 @@ export class PaymentPlan {
       this.instalmentAmount = input.instalmentAmount
       this.firstPaymentDate = new LocalDate().deserialize(input.firstPaymentDate)
       this.paymentSchedule = input.paymentSchedule ? PaymentSchedule.of(input.paymentSchedule.value) : undefined
+      this.completionDate = new LocalDate().deserialize(input.completionDate)
+      this.paymentLength = input.paymentLength
     }
     return this
   }
+
+  getCompletionDate (): LocalDate {
+    const paymentPlan: PaymentPlanCalcs = PaymentPlanCalcs.create(this.totalAmount, this.instalmentAmount, Frequency.of(this.paymentSchedule.value), this.firstPaymentDate.toMoment())
+    const lastPaymentDate = paymentPlan.calculateLastPaymentDate()
+    return new LocalDate(lastPaymentDate.year(), lastPaymentDate.month(), lastPaymentDate.date())
```

Comment:
`lastPaymentDate.month()` from `moment.js` is zero index. it will be one month off.